tools/elf2efi.lds

```diff
@@ -6,6 +6,11 @@ SECTIONS {
         .text ALIGN(CONSTANT(MAXPAGESIZE)) : {
                 *(.text .text.*)
         }
+
+        /* When linking a minimal addon stub the linker can merge .text and .dynsym creating a RWE segment,
+         * and then rejects it. Ensure there's a gap. */
+        . = ALIGN (CONSTANT (COMMONPAGESIZE));
```

Comment:
hmm, can you explain to me in more detail what this does? i don't get it, too dumb

Reply:
The linker creates a ELF load segment that contains code (rx) and data (rw), which it doesn't like. This just moves the data segment to a new page so that the code segments don't have (rwx) permissions.

I think this happens because the add-on ELF binary does not contain any of the other sections for which we give an explicit alignment already. So we have to explicitly move the current address so that the other sections following don't end up in the same segment.

Might be better to change the whole script to use this style instead of putting the alignment as part of the section description...:
```
. = ALIGN(CONSTANT(MAXPAGESIZE));
.section : {}
```

(Also, don't get confused by the fact that PE does not have a notion of segments. This is a side-effect of not having native PE builds and using ELF to PE conversion instead.)

Reply:
but why does explicit alignment force separations of segments? if the segments happen to be aligned already then linker might still merge them? what am i missing?

Reply:
Assigning to the current address will always advance it:
```
. = EXPR;
```
While using the address on a section will only advance the current address *if* this section exists / will be created:
```
 .section EXPR : {}
```

Right now we have this:
```
.text ALIGN(...) : {...}
.rodata ALIGN(...) : {...}
...
.dynsym : {...}
```

Because the sections in between `.text` and `.dynsym` are missing in this minimal binary, the linker puts `.dynsym` in the same segment.

Reply:
ah, might be good to include a short comment along those lines into the file, for the casual onlooker.